tests/rainbowGrades/test_csci1100.py

```diff
@@ -0,0 +1,295 @@
+
+    if output_generated_contents != output_known_contents:
+        error_and_cleanup(test_tmp, "Generated output.html did not match expected output.html")
+
+    # Extract test version of individual_grade_summary_html files
+    known_individual_path = os.path.join(test_tmp, "individual_summary_html")
+    summary_individual_path = os.path.join(summary_tmp, "individual_summary_html")
+    os.mkdir(known_individual_path)
+    return_code = subprocess.call(["tar", "-xf", os.path.join(script_path, "individual_summary_10090542_csci1100.tar"),
+                     "-C", known_individual_path])
+    if return_code != 0:
+        error_and_cleanup(test_tmp, "Extracting raw data failed (Error {}".format(return_code))
+
+    # Construct lists of generated and test individual_grade_summary_html files
+    known_files = []
+    summary_files = []
+    for f in os.listdir(known_individual_path):
+        known_files.append(f)
+    for f in os.listdir(summary_individual_path):
+        summary_files.append(f)
+
+    # Check that the name and number of individual_grade_summary_html files are the same
+    if len(known_files) != len(summary_files):
+        file_diff = len(known_files) - len(summary_files)
+        if len(known_files) > len(summary_files):
+            error_and_cleanup(test_tmp, "There are {} more files in the generated ".format(file_diff) +
+                              "individual_summary_html than expected.")
+        else:
+            error_and_cleanup(test_tmp, "There are {} fewer files in the generated ".format(-1*file_diff) +
+                              "individual_summary_html than expected.")
+
+    # Compare the contents (excluding time-sensitive parts) of generated and test individual_grade_summary_html files
+    # TODO: Currently not checking generated personal message files (for seating/materials)
+    for f in known_files:
+        # Skip any files that don't end in summary.html (i.e. messages.json/.html)
+        if f[-12:] != "summary.html":
+            continue
+        filename1 = os.path.join(known_individual_path, f)
+        filename2 = os.path.join(summary_individual_path, f)
+        contents1 = ""
+        contents2 = ""
+
+        try:
+            with open(filename1, 'r') as file1:
+                contents1 = file1.readlines()
+
+            with open(filename2, 'r') as file2:
+                contents2 = file2.readlines()
+        except Exception as e:
+            error_and_cleanup(test_tmp, "{}".format(e))
+
+        # Construct and use filters to ignore time-dependent contents during comparison
+        filter1 = filter(remove_info_last_updated, contents1)
+        filter2 = filter(remove_info_last_updated, contents2)
+        same_flag = True
+        for x, y in zip(filter1, filter2):
+            if x != y:
+                same_flag = False
+        if not same_flag:
+            error_and_cleanup(test_tmp, "{} and {} differ".format(filename1, filename2))
```

Comment:
Same as above, do we really need to wait till we've read through the entirety of `zip(filter1, filter2)` before erroring?